src/app/resources/view-resources/resources-view.component.ts

```diff
@@ -91,7 +97,79 @@ export class ResourcesViewComponent implements OnInit, OnDestroy {
     this.couchSrc = this.urlPrefix + resource._id + '/' + filename;
   }
 
-  resource_activity(resourceId, activity) {
+  getResourceRating(resource_id) {
+    this.couchService
+      .post('ratings/_find', findDocuments({ 'item': resource_id, 'type': 'resource' }, 0 ))
+      .subscribe((ratings) => {
+        // Counts number of ratings, number of male/female ratings, adds the total rating sum,
+        // and gets the logged in user's rating if applicable.
+        const { rateSum, userRating, femaleCount, maleCount, totalCount } = ratings.docs.reduce((stats, rating) => {
+          stats.userRating = (rating.user.name === this.userService.get().name) ? rating : stats.userRating;
+          stats.totalCount++;
+          switch (rating.user.gender) {
+            case 'male':
+              stats.maleCount++;
+              break;
+            case 'female':
+              stats.femaleCount++;
+              break;
+          }
+          stats.rateSum = stats.rateSum + parseInt(rating.rate, 10);
+          return stats;
+        }, { rateSum: 0, userRating: '', totalCount: 0, maleCount: 0, femaleCount: 0 });
+        Object.assign(this.rating, {
+          femalePercent: femaleCount === 0 ? 0 : ((femaleCount / totalCount) * 100).toFixed(0),
+          malePercent: maleCount === 0 ? 0 : ((maleCount / totalCount) * 100).toFixed(0),
+          average: totalCount === 0 ? 0 : rateSum / totalCount,
+          userRating, totalCount });
+      }, error => console.log(error));
+  }
+
+  openRatingDialog(resource_id) {
+    const title = 'Rating';
+    const type = 'rating';
+    const fields =
+      [
+        { 'label': 'Rate', 'type': 'radio', 'name': 'rate', 'placeholder': 'Your Rating', 'required': false },

```

Comment:
can we add some extra parameter so it says its rating and form only change for rating not all radio